# Understanding Homelessness Rates

## Table of Contents
<ul>
<li><a href="#intro">Introduction</a></li>
<li><a href="#wrangling">Data Wrangling</a></li>
<li><a href="#eda">Exploratory Data Analysis</a></li>
<li><a href="#conclusions">Conclusions</a></li>
</ul>

## Introduction
**Need to add info about the databases and assumptions of homelessness including personal v macro issues**

### Data Background
- Originally was going to use a dataset from Kaggle but decided to pull straight from HUD-CoC site. 
- Used [2007 - 2017 PIT Counts by State](https://www.hudexchange.info/resource/3031/pit-and-hic-data-since-2007/) and converted to single database

(borrowed from: https://www.kaggle.com/bltxr9/eda-of-total-homeless-population)
This dataset was generated by CoC and provided to HUD. Note: HUD did not conduct a full data quality review on the data submitted by each CoC.

What is the [Continuum of Care (CoC) Program](https://www.hudexchange.info/programs/coc/)?

Original Data: [PIT and HIC Data Since 2007](https://www.hudexchange.info/resource/3031/pit-and-hic-data-since-2007/)

CoC-HUD Summary Reports: [CoC Homeless Populations and Subpopulations Reports](https://www.hudexchange.info/programs/coc/coc-homeless-populations-and-subpopulations-reports/)

**Other Resources**

[Funding Awards](https://www.hudexchange.info/programs/coc/awards-by-component/)

[CoC Dashboard Reports](https://www.hudexchange.info/programs/coc/coc-dashboard-reports/)

[CoC Housing Inventory Count Reports](https://www.hudexchange.info/programs/coc/coc-housing-inventory-count-reports/)

## Data Wrangling

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
%matplotlib inline

### Inspect Data

In [72]:
df = pd.read_csv('homeless-pit-by-state.csv')
df.head()

State  Year  Number of CoCs Total Homeless Sheltered Homeless  \
0    AK  2017               2          1,845              1,551   
1    AL  2017               8          3,793              2,656   
2    AR  2017               6          2,467              1,273   
3    AZ  2017               3          8,947              5,781   
4    CA  2017              43        134,278             42,636   

  Unsheltered Homeless Homeless Individuals Sheltered Homeless Individuals  \
0                  294                1,354                          1,060   
1                1,137                2,985                          1,950   
2                1,194                2,068                            937   
3                3,166                6,488                          3,423   
4               91,642              112,756                         25,022   

  Unsheltered Homeless Individuals Homeless People in Families  \
0                              294                         491   
1                            1,035                         808   
2                            1,131                         399   
3                            3,065                       2,459   
4                           87,734                      21,522   

                    ...                    \
0                   ...                     
1                   ...                     
2                   ...                     
3                   ...                     
4                   ...                     

  Unsheltered Parenting Youth (Under 25) Parenting Youth Under 18  \
0                                      0                        0   
1                                      3                        6   
2                                      0                        0   
3                                      0                        0   
4                                    234                       16   

  Sheltered Parenting Youth Under 18 Unsheltered Parenting Youth Under 18  \
0                                  0                                    0   
1                                  6                                    0   
2                                  0                                    0   
3                                  0                                    0   
4                                 11                                    5   

  Parenting Youth Age 18-24 Sheltered Parenting Youth Age 18-24  \
0                        22                                  22   
1                        23                                  20   
2                        10                                  10   
3                        81                                  81   
4                       874                                 645   

  Unsheltered Parenting Youth Age 18-24 Children of Parenting Youth  \
0                                     0                          39   
1                                     3                          39   
2                                     0                          13   
3                                     0                         112   
4                                   229                       1,058   

  Sheltered Children of Parenting Youth  \
0                                    39   
1                                    35   
2                                    13   
3                                   112   
4                                   782   

  Unsheltered Children of Parenting Youth  
0                                       0  
1                                       4  
2                                       0  
3                                       0  
4                                     276  

[5 rows x 45 columns]

#### Check Data Types and Missing Data

In [11]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 605 entries, 0 to 604
Data columns (total 45 columns):
State                                                          605 non-null object
Year                                                           605 non-null int64
Number of CoCs                                                 605 non-null int64
Total Homeless                                                 605 non-null object
Sheltered Homeless                                             605 non-null object
Unsheltered Homeless                                           605 non-null object
Homeless Individuals                                           605 non-null object
Sheltered Homeless Individuals                                 605 non-null object
Unsheltered Homeless Individuals                               605 non-null object
Homeless People in Families                                    605 non-null object
Sheltered Homeless People in Families                          605 

_Observations_
- Missing data is consistent across groups of categories. Visual inspection of data confirms that this is due to additional categories added in subsequent years. 
- Data is all in object format and will need to be converted to int. 
- Not all categories are multually exclusive, confirmation is required to confirm how data is summed.

Visualization of data that all columns are available from 2015 onwards, 2011 - 2014 contains columns up to Unsheltered Homeless Veterans and 2007 - 2013 contains columns up to Unsheltered Chronically Homeless Individuals.

#### Check unique values

In [12]:
df.nunique()

State                                                           56
Year                                                            11
Number of CoCs                                                  34
Total Homeless                                                 577
Sheltered Homeless                                             570
Unsheltered Homeless                                           523
Homeless Individuals                                           573
Sheltered Homeless Individuals                                 560
Unsheltered Homeless Individuals                               520
Homeless People in Families                                    558
Sheltered Homeless People in Families                          550
Unsheltered Homeless People in Families                        378
Chronically Homeless                                           522
Sheltered Chronically Homeless                                 461
Unsheltered Chronically Homeless                              

_Observations_
- There are 11 years of data contained in the set (2007 - 2017)
- Need to confirm what states are covered within state

In [13]:
df['State'].unique()

array(['AK', 'AL', 'AR', 'AZ', 'CA', 'CO', 'CT', 'DC', 'DE', 'FL', 'GA',
       'GU', 'HI', 'IA', 'ID', 'IL', 'IN', 'KS', 'KY', 'LA', 'MA', 'MD',
       'ME', 'MI', 'MN', 'MO', 'MP', 'MS', 'MT', 'NC', 'ND', 'NE', 'NH',
       'NJ', 'NM', 'NV', 'NY', 'OH', 'OK', 'OR', 'PA', 'PR', 'RI', 'SC',
       'SD', 'TN', 'TX', 'UT', 'VA', 'VI', 'VT', 'WA', 'WI', 'WV', 'WY',
       'KS*'], dtype=object)

- It appears that the list includes some US territories and DC. I am less familiar with state abbreviations - will need to get state names for reference.
- One of states is KS*, note from dataset says: The number of CoCs in 2017 was 399. However, MO-604 merged in 2016 and covers territory in both MO and KS, contributing to the PIT count in both states. This will need to be inspected individually to understand.

### Clean Data

In [64]:
df.columns

Index(['State', 'Year', 'Number of CoCs', 'Total Homeless',
       'Sheltered Homeless', 'Unsheltered Homeless', 'Homeless Individuals',
       'Sheltered Homeless Individuals', 'Unsheltered Homeless Individuals',
       'Homeless People in Families', 'Sheltered Homeless People in Families',
       'Unsheltered Homeless People in Families', 'Chronically Homeless',
       'Sheltered Chronically Homeless', 'Unsheltered Chronically Homeless',
       'Chronically Homeless Individuals',
       'Sheltered Chronically Homeless Individuals',
       'Unsheltered Chronically Homeless Individuals',
       'Chronically Homeless People in Families',
       'Sheltered Chronically Homeless People in Families',
       'Unsheltered Chronically Homeless People in Families',
       'Homeless Veterans', 'Sheltered Homeless Veterans',
       'Unsheltered Homeless Veterans',
       'Homeless Unaccompanied Youth (Under 25)',
       'Sheltered Homeless Unaccompanied Youth (Under 25)',
       'Unsheltered Home

In [66]:
df['Year'] = df['Year'].astype(int)

In [68]:
df['Number of CoCs'] = df['Number of CoCs'].astype(int)

In [80]:
df['Total Homeless'] = df['Total Homeless'].apply(lambda x: x.replace(',', '')).astype(float)

ValueError: could not convert string to float: '.'

In [78]:
string = 'sdfs0sdfs0'
string.replace('0', '')

'sdfssdfs'